### PR TITLE
Yoshi 4 migration ooi template

### PR DESCRIPTION
**Pull request by ranyitz**
### 🔦 Summary

Moving the ooi templates to use yoshi 4

### 🐾 step

__JavaScript__
* [x] yoshi 4
* [x] App Flow

__TypeScript__
* [x] yoshi 4
* [x] App Flow

### CODE REVIEW

**Comment by ranyitz** [comment]
@gileck The changes above update the dependencies in the Yoshi but also separate the `dev` code from the `test` code, which I think makes a lot of sense.

It should give a cleaner behavior for `ooi` app developers.

Please review and give your opinion on them.

File: packages/create-yoshi-app/templates/out-of-iframe/javascript/jest-yoshi.config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   server: {
-    command: 'node test/fake-server.js',
+    command: 'node dev/server.js',
```
Comment (Schniz) [inline packages/create-yoshi-app/templates/out-of-iframe/javascript/jest-yoshi.config.js]: What about #1259?

Comment (Schniz) [reply packages/create-yoshi-app/templates/out-of-iframe/javascript/jest-yoshi.config.js]: maybe we should support both `index-dev.js` and `dev/server.js`
1. what precedence?
2. what about typescript?

File: packages/create-yoshi-app/templates/out-of-iframe/javascript/dev/server.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ const fs = require('fs');
 const server = httpTestkit.server({
   port: process.env.PORT,
   ssl: {
-    cert: fs.readFileSync('./test/certificates/cert.pem', 'utf-8'),
-    key: fs.readFileSync('./test/certificates/key.pem', 'utf-8'),
+    cert: fs.readFileSync('dev/certificates/cert.pem', 'utf-8'),
```
Comment (Schniz) [inline packages/create-yoshi-app/templates/out-of-iframe/javascript/dev/server.js]: `path.resolve(__dirname, ...)`

File: packages/create-yoshi-app/templates/out-of-iframe/typescript/dev/server.js [hunk 3]
```diff
@@ -7,8 +7,8 @@ const fs = require('fs');
 const server = httpTestkit.server({
   port: process.env.PORT,
   ssl: {
-    cert: fs.readFileSync('./test/certificates/cert.pem', 'utf-8'),
-    key: fs.readFileSync('./test/certificates/key.pem', 'utf-8'),
+    cert: fs.readFileSync('dev/certificates/cert.pem', 'utf-8'),
```
Comment (Schniz) [inline packages/create-yoshi-app/templates/out-of-iframe/typescript/dev/server.js]: `path.resolve(__dirname, ...)`

**Pull request merged**